test/e2e/apps/statefulset.go

```diff
@@ -776,7 +776,6 @@ var _ = SIGDescribe("StatefulSet", func() {
 			ss, err := c.AppsV1().StatefulSets(ns).Create(ss)
 			framework.ExpectNoError(err)
 			e2esset.WaitForRunningAndReady(c, *ss.Spec.Replicas, ss)
-			ss = e2esset.WaitForStatus(c, ss)
```

Comment:
Even though it should not happen that we block here, I'd not remove this. If this starts failing it'll be a signal we removed important part of the controller code which is responsible for updating `ObservedGeneration`.

Reply:
how about 
```
e2esset.WaitForStatus(c, ss)
```
just delete `ss = `